### PR TITLE
Qt: fix deprecation warnings

### PR DESCRIPTION
**Pull request by oltolm**

### CODE REVIEW

**Comment by C0rn3j** [comment]
Modern Qt without this PR:
```
/build/shadps4-git/src/shadps4/src/qt_gui/settings_dialog.cpp:133:53: warning: ‘void QCheckBox::stateChanged(int)’ is deprecated: Use checkStateChanged() instead [-Wdeprecated-declarations]
  133 |         connect(ui->fullscreenCheckBox, &QCheckBox::stateChanged, this,
```

EOL Qt with this PR:
```
/home/runner/work/shadPS4/shadPS4/src/qt_gui/settings_dialog.cpp:273:47: error: no member named 'checkStateChanged' in 'QCheckBox'
  273 |         connect(ui->rdocCheckBox, &QCheckBox::checkStateChanged, this,
```

`checkStateChanged()` is only present in Qt 6.7 and up.

`stateChanged()` will be last present in Qt 6.9 (or removed in 6.9, I can't parse what "until 6.9" means) - [releasing in about 3 months](https://wiki.qt.io/Qt_6.9_Release) - `Final 18.03.2025`.

https://doc.qt.io/qt-6/qcheckbox.html#signals

Noble, which the current build system depends on, [uses Qt 6.4](https://answers.launchpad.net/ubuntu/noble/+package/qt6-base-dev).

6.4 has been [EOL since 2023-03-31](https://endoflife.date/qt).

Switching the CI to Ubuntu 24.10 (Oracular) won't help, as while it does ship newer Qt, it's still an old EOL version without support [at 6.6](https://launchpad.net/ubuntu/oracular/+package/qt6-base-dev)

25.04 (which should release 2025-05-xx) should work fine, but that's 5 months away, though I don't really see an alternative unless you want to somehow ifdef it for different Qt versions (is that even doable?).

TL;DR this either has to:
* wait a very long time
* build system needs large changes
* (ideal if possible) ifdef for qt versions needs to be added

EDIT: Conflicts after https://github.com/shadps4-emu/shadPS4/commit/3f1061de5613c0c4a74d6394a6493491280bc03f

**Comment by oltolm** [comment]
I updated it.

**Comment by Hermiten** [comment]
Is it still up to date?

**Comment by C0rn3j** [comment]
Looks like it, but it also still misses any Qt version check ifdefs, as per my previous message, so can't be pulled as is.

**Comment by oltolm** [comment]
I added ifdefs.

File: src/qt_gui/check_update.cpp [hunk 1]
```diff
@@ -254,7 +253,11 @@ void CheckUpdate::setupUI(const QString& downloadUrl, const QString& latestDate,
     connect(noButton, &QPushButton::clicked, this, [this]() { close(); });
 
     autoUpdateCheckBox->setChecked(Config::autoUpdate());
+#if (QT_VERSION < QT_VERSION_CHECK(6, 8, 0))
```
Comment (C0rn3j) [inline src/qt_gui/check_update.cpp]: Shouldn't these two checks be <6.7.0?

https://code.qt.io/cgit/qt/qtreleasenotes.git/about/qt/6.7.0/release-note.md

```
5a96d13bb5 QCheckBox: add new checkStateChanged(Qt::CheckState) signal
Added new checkStateChanged(Qt::CheckState) signal, obsoleting stateChanged(int).
```

File: src/qt_gui/settings_dialog.cpp [hunk 2]
```diff
@@ -130,8 +130,13 @@ SettingsDialog::SettingsDialog(std::span<const QString> physical_devices,
     // GENERAL TAB
     {
 #ifdef ENABLE_UPDATER
+#if (QT_VERSION < QT_VERSION_CHECK(6, 8, 0))
```
Comment (C0rn3j) [inline src/qt_gui/settings_dialog.cpp]: As above

File: src/qt_gui/check_update.cpp (hunk 1 above)
Comment (oltolm) [reply src/qt_gui/check_update.cpp]: Yes, I will fix it.

File: src/qt_gui/settings_dialog.cpp (hunk 2 above)
Comment (oltolm) [reply src/qt_gui/settings_dialog.cpp]: Fixed.

File: src/qt_gui/check_update.cpp (hunk 1 above)
Comment (oltolm) [reply src/qt_gui/check_update.cpp]: Fixed.
